vllm/v1/worker/gpu_model_runner.py

```diff
@@ -604,6 +607,54 @@ def profile_run(self) -> None:
             torch.tensor([], dtype=torch.float32, device=self.device)
             for _ in range(self.num_attn_layers)
         ]
+
+        # Profile with multimodal encoder & encoder cache.
+        # TODO (ywang96): generalize this beyond image modality since
+        # mm_input_mapper only supports image inputs.
+        if self.is_multimodal_model:
+
+            # Create dummy batch of multimodal inputs.
+            dummy_mm_data = self.input_registry \
+                .dummy_data_for_profiling(self.model_config,
+                                            self.max_num_tokens,
+                                            self.mm_registry).multi_modal_data
+            dummy_mm_kwargs, _ = self.mm_input_mapper.process_inputs(
+                mm_data=dummy_mm_data,
+                mm_hashes=None,
+                mm_processor_kwargs=None,
+                precomputed_mm_inputs=None)
+
+            # NOTE: Currently model is profiled with a single non-text
+            # modality even when it supports multiple.
+            max_tokens_per_mm_item = max(
+                self.mm_registry.get_max_tokens_per_item_by_modality(
+                    self.model_config).values())
+
+            max_num_mm_items = min(
+                self.max_num_encoder_input_tokens,
+                self.encoder_cache_size) // max_tokens_per_mm_item
+
+            # Dummy data definition in V0 may contain multiple items for
+            # a single request, therefore here we always replicate first
+            # item by max_num_mm_items times.
```

Comment:
Can you elaborate more? What does "multiple items" mean here? Does it mean multiple images?

Reply:
Yes, the dummy data in a dummy sequence for an LMM is based on `limit-mm-per-prompt`. This means if the user sets up a engine with `limit-mm-per-prompt image=2`, then the dummy data for a single sequence will contain two identical dummy images. We had to do this because in V0 encoder call is scheduled together with decoder call.

I will modify the comment here to clarify.